flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -227,4 +235,28 @@ private boolean validateDeployment(FlinkResourceContext<FlinkDeployment> ctx) {
         }
         return true;
     }
+
+    private void setCRStatus(FlinkDeployment flinkApp) {
+        final List<Condition> conditions = new ArrayList<>();
+        FlinkDeploymentStatus deploymentStatus = flinkApp.getStatus();
+        switch (deploymentStatus.getJobManagerDeploymentStatus()) {
```

Comment:
I think we need a bit more logic here. We need to check if the status doesn't already have a condition in it's list that matches the same type, we need to then check the reason and message of this condition and only if they are different replace the existing condition with the new one. 

We shouldn't just replace the existing list of conditions with a new one, there may be additional conditions of different types added later that would get removed. This may be one of the reasons why the tests have had to be modified to bump up the number of status changes

Reply:
Modified logic to update the conditions.